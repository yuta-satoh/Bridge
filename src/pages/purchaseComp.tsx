import Image from "next/image";
import Head from "next/head";
import Link from "next/link";
import pstyles from "../styles/purchaseComp.module.css"
import React from "react";

export default function purchaseComp(){

return(
    <>
    <Head>
        <title>購入完了</title>
    </Head>
    <main className={pstyles.main}>
        <div className={pstyles.img}>
        <Link href="/">
        <Image
        src="/purchase3.jpeg"
        alt="購入完了"
        width={650}
        height={400}
        />
        </Link>
        </div>

        <div className={pstyles.order}>
            <div className={pstyles.order_number}>
            <ul>
            <li>お客様のご注文番号</li>
            <li className={pstyles.order_num}>1234456ランダムで入る</li>
            </ul>
            </div>

            <div className={pstyles.order_details}>
            <ul>
            <li>ご注文ありがとうございます。</li>
            <li>明細メールをご登録メールアドレスにお送りしましたのでご確認ください。</li>
            </ul>
            </div>
        </div>

        <div className={pstyles.order_toplink}>
        <Link href="/">
            TOPページへ→
            </Link>
        </div>
    </main>
    </>

)

}
