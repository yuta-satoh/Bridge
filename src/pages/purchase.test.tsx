import renderer from 'react-test-renderer';
import { Middleware } from 'swr';
import { SWRResponse } from 'swr';
import { SWRConfig } from 'swr';
import Purchase from './purchase';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import purchaseComp from './purchaseComp';

type User = {
  lastName: string;
  firstName: string;
  gender: string;
  tell: string;
  email: string;
  zipcode: string;
  address: string;
  password: string;
}[];

type items = {
  id: number;
  name: string;
  description: string;
  genre: string;
  category: string;
  price: number;
  imgurl: string;
  stock: number;
  delete: boolean;
};
type cart = {
  id: number;
  item_id: number;
  items: items;
  cart_id: number;
  date: string;
  quantity: number;
  delete: boolean;
}[];

const user: User = [
  {
    lastName: '楽須',
    firstName: '太郎',
    gender: 'male',
    tell: '111-1111-1111',
    email: 'rakus@example.com',
    zipcode: '160-0022',
    address: '東京都新宿区新宿2-5-12 8F',
    password: 'rakus',
  },
];

jest.mock('next/router', () => ({
  useRouter() {
    return {
      push: jest.fn(),
    };
  },
}));

const fetchMock = () => {
  return new Promise((resolve) => {
    resolve({
      status: 200,
    });
  });
};

test('描画される', async () => {
  const cartItem: cart = [
    {
      id: 1,
      item_id: 1,
      items: {
        id: 1,
        name: 'ベルベットアームチェア',
        description: 'ベルベットのお洒落なデザインチェア。',
        genre: 'フェミニン',
        category: '椅子',
        price: 16000,
        imgurl: '/images/chair/chair_feminine_1.jpg',
        stock: 72,
        delete: false,
      },
      cart_id: 1,
      date: '2023-03-01',
      quantity: 1,
      delete: false,
    },
  ];
  const testMiddleware: Middleware = () => {
    return (): SWRResponse<any, any> => {
      return {
        data: cartItem,
        error: undefined,
        mutate: (_) => Promise.resolve(),
        isValidating: false,
        isLoading: false,
      };
    };
  };
  const tree = renderer
    .create(
      <SWRConfig value={{ use: [testMiddleware] }}>
        <Purchase cookie={'1'} user={user} />
      </SWRConfig>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('描画されない', async () => {
  Object.defineProperty(window, 'location', {
    value: {
      href: '/cart',
      pathname: '/cart',
      search: '',
      hostname: '',
    },
  });
  const cartItem: cart = [];
  const testMiddleware: Middleware = () => {
    return (): SWRResponse<any, any> => {
      return {
        data: cartItem,
        error: undefined,
        mutate: (_) => Promise.resolve(),
        isValidating: false,
        isLoading: false,
      };
    };
  };
  const tree = renderer.create(
    <SWRConfig value={{ use: [testMiddleware] }}>
      <Purchase cookie={'1'} user={user} />
    </SWRConfig>
  );
  expect(window.location.href).toEqual('/cart');
});

test('送信される', async () => {
  const mockRouter = {
    push: jest.fn(),
  };
  global.fetch = jest.fn().mockImplementation(fetchMock);
  const cartItem: cart = [
    {
      id: 1,
      item_id: 1,
      items: {
        id: 1,
        name: 'ベルベットアームチェア',
        description: 'ベルベットのお洒落なデザインチェア。',
        genre: 'フェミニン',
        category: '椅子',
        price: 16000,
        imgurl: '/images/chair/chair_feminine_1.jpg',
        stock: 72,
        delete: false,
      },
      cart_id: 1,
      date: '2023-03-01',
      quantity: 1,
      delete: false,
    },
  ];
  const testMiddleware: Middleware = () => {
    return (): SWRResponse<any, any> => {
      return {
        data: cartItem,
        error: undefined,
        mutate: (_) => Promise.resolve(),
        isValidating: false,
        isLoading: false,
      };
    };
  };
  render(
    <SWRConfig value={{ use: [testMiddleware] }}>
      <Purchase cookie={'1'} user={user} />
    </SWRConfig>
  );
  const submitButton = screen.getByText('購入する');
  //   expect(submitButton).toMatchSnapshot()
  await userEvent.click(submitButton);
  //   expect(mockRouter.push).toEqual('/purchaseComp');
});
