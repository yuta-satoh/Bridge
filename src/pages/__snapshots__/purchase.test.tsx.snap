// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`描画される 1`] = `
<div
  className="body"
>
  <h1
    className="title"
  >
    購入確認
  </h1>
  <table
    className="itemTable"
  >
    <tbody>
      <tr
        className="tableTLine"
      >
        <th
          className="nameTag"
        >
          品名
        </th>
        <th
          className="item"
        >
          個数
        </th>
        <th
          className="itemTag"
        >
          単価
        </th>
        <th
          className="itemTag"
        >
          小計
        </th>
      </tr>
      <tr
        className="tableLine"
      >
        <td
          className="itemName"
        >
          ベルベットアームチェア
        </td>
        <td
          className="item"
        >
          1
        </td>
        <td
          className="item"
        >
          ¥ 
          16,000
        </td>
        <td
          className="item"
        >
          ¥
           
          16,000
        </td>
      </tr>
    </tbody>
    <tbody>
      <tr
        className="content"
      >
        <td
          colSpan={2}
        />
        <td
          className="subTTotal"
        >
          本体合計
        </td>
        <td
          className="topText"
        >
          ¥ 
          16,000
        </td>
      </tr>
      <tr
        className="content"
      >
        <td
          colSpan={2}
        />
        <td
          className="subTotal"
        >
          消費税
        </td>
        <td
          className="text"
        >
          ¥ 
          1,600
        </td>
      </tr>
      <tr
        className="totalContent"
      >
        <td
          colSpan={2}
        />
        <td
          className="total"
        >
          合計
        </td>
        <td
          className="total"
        >
          ¥ 
          17,600
        </td>
      </tr>
    </tbody>
  </table>
  <div
    className="addressArea"
  >
    <p
      className="addressTitle"
    >
      お届け先住所
    </p>
    <div
      className="radio"
    >
      <input
        checked={true}
        id="now"
        name="addressSelect"
        onChange={[Function]}
        type="radio"
        value="false"
      />
      <label
        className="label"
        htmlFor="now"
      >
        現在の住所にお届け
      </label>
    </div>
    <p
      className="nAddress"
    >
      東京都新宿区新宿2-5-12 8F
    </p>
    <div
      className="radio"
    >
      <input
        id="new"
        name="addressSelect"
        onChange={[Function]}
        type="radio"
        value="true"
      />
      <label
        className="label"
        htmlFor="new"
      >
        別の住所にお届け
      </label>
    </div>
    <div
      className="nonItems"
    >
      <label
        className="label"
        htmlFor="zipcode"
      >
        郵便番号
      </label>
      <br />
      <input
        className="zipcode border 'border-neutral-500' rounded pl-2.5"
        id="zipcode1"
        name="zipcode1"
        onChange={[Function]}
        pattern="^[0-9]+$"
        type="text"
        value=""
      />
      <span>
        -
      </span>
      <input
        className="zipcode border 'border-neutral-500' rounded pl-2.5"
        id="zipcode2"
        name="zipcode2"
        onChange={[Function]}
        pattern="^[0-9]+$"
        type="text"
        value=""
      />
      <button
        className="zipButton text-white bg-neutral-900 border border-neutral-900 rounded px-1"
        onClick={[Function]}
        type="button"
      >
        住所検索
      </button>
    </div>
    <div
      className="nonItems"
    >
      <label
        className="label"
        htmlFor="address"
      >
        住所
      </label>
      <br />
      <input
        className="address border 'border-neutral-500' rounded pl-2.5"
        id="address"
        name="address"
        onChange={[Function]}
        type="text"
        value="東京都新宿区新宿2-5-12 8F"
      />
    </div>
  </div>
  <div
    className="dateArea"
  >
    <p
      className="addressTitle"
    >
      お届け希望日
    </p>
    <input
      className="date"
      id="date"
      min="2023-03-21"
      name="date"
      type="date"
    />
  </div>
  <div
    className="timeArea"
  >
    <p
      className="addressTitle"
    >
      希望時間
    </p>
    <select
      className="time_zone"
      id="time_zone"
      name="time_zone"
    >
      <option
        value="none"
      >
        希望無し
      </option>
      <option
        value="morning"
      >
        午前
      </option>
      <option
        value="noon"
      >
        12時~18時
      </option>
      <option
        value="evening"
      >
        18時以降
      </option>
    </select>
  </div>
  <div
    className="buttonArea"
  >
    <button
      className="buttonStyle"
      onClick={[Function]}
      type="button"
    >
      購入する
    </button>
  </div>
  <div
    className="loginLink"
  >
    <a
      href="/cart"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <button
        className="linkButton"
        type="button"
      >
        カートに戻る
        <span
          className="buttonSpan"
        >
          →
        </span>
      </button>
    </a>
  </div>
</div>
`;
